filebeat/crawler/prospector.go

```diff
@@ -102,6 +102,14 @@ func (p *Prospector) Stop() {
 	close(p.done)
 }
 
+func (p *Prospector) AddHarvester(file string, stat *harvester.FileStat) (*harvester.Harvester, error) {
```

Comment:
Kind of, but it is not unique as a new file with the same file name can appear. I didn't make any changes here yet. Currently not identifier for a "harvester" is needed, only for the files itself.